src/gen/conditions.rs

```diff
@@ -3311,3 +3371,119 @@ fn test_multiple_my_birth_assertions(
 
     test(spend);
 }
+
+#[test]
+fn test_assert_ephemeral() {
+    // ASSERT_EPHEMERAL
+    // the coin11 value is the coinID computed from (H1, H1, 123). The first
+    // coin we spend in this case
+    // 51 is CREATE_COIN, 76 is ASSERT_EPHEMERAL
+    let test = "(\
+       (({h1} ({h1} (123 (\
+           ((51 ({h2} (123 ) \
+           ))\
+       (({coin11} ({h2} (123 (\
+           ((76 ) \
+           ))\
+       ))";
+    // we don't expect any error
+    let (a, conds) = cond_test(test).unwrap();
+
+    // just make sure there are no constraints
+    assert_eq!(conds.agg_sig_unsafe.len(), 0);
+    assert_eq!(conds.reserve_fee, 0);
+    assert_eq!(conds.cost, CREATE_COIN_COST);
```

Comment:
These 3 lines would make a great named test function if you use them elsewhere.

Reply:
well, they aren't really that common. The comment is a bit misleading, suggesting that these lines ensure there are no constraints. I'll remove it.